pennylane/transforms/qcut.py

```diff
@@ -287,6 +294,85 @@ def fragment_graph(graph: MultiDiGraph) -> Tuple[Tuple[MultiDiGraph], MultiDiGra
     return subgraphs, communication_graph
 
 
+def _find_new_wire(wires: Wires) -> int:
+    """Finds a new wire label that is not in ``wires``."""
+    ctr = 0
+    while ctr in wires:
+        ctr += 1
+    return ctr
+
+
+# pylint: disable=protected-access
+def graph_to_tape(graph: MultiDiGraph) -> QuantumTape:
+    """
+    Converts a directed multigraph to the corresponding :class:`~.QuantumTape`.
+
+    Each node in the graph should have an order attribute specifying the topological order of
+    the operations. This maintains ordering of operations and allows for the
+    deferred measurement principle to be applied when necessary.
+
+    .. note::
+
+        This function is designed for use as part of the circuit cutting workflow.
+        Check out the :doc:`transforms </code/qml_transforms>` page for more details.
+
+    Args:
+        graph (MultiDiGraph): directed multigraph to be converted to a tape
+
+    Returns:
+        QuantumTape: the quantum tape corresponding to the input graph
+
+    **Example**
+
+    Consider the following, where ``graph`` contains three :class:`~.MeasureNode` and
+    :class:`~.PrepareNode` pairs that divide the full circuit graph into five subgraphs.
+    We can find the circuit fragments by using:
+
+    .. code-block:: python
+
+        >>> subgraphs, communication_graph = qml.transforms.fragment_graph(graph)
+        >>> tapes = [qml.transforms.graph_to_tape(sg) for sg in subgraphs]
+        >>> tapes
+        [<QuantumTape: wires=[0], params=1>, <QuantumTape: wires=[0, 1], params=1>,
+         <QuantumTape: wires=[1], params=1>, <QuantumTape: wires=[0], params=0>,
+         <QuantumTape: wires=[1], params=0>]
+    """
+
+    wires = Wires.all_wires([n.wires for n in graph.nodes])
+
+    ordered_ops = sorted(
+        [(order, op) for op, order in graph.nodes(data="order")], key=lambda x: x[0]
+    )
+    wire_map = {w: w for w in wires}
+
+    copy_ordered_ops = copy.deepcopy(ordered_ops)
+
+    updated = {}
+    with QuantumTape() as tape:
+        for _, op in copy_ordered_ops:
+            name_and_wire = op.name + str(op.wires)
+
+            new_wires = [wire_map[w] for w in op.wires]
+            op._wires = Wires(new_wires)  # TODO: find a better way to update operation wires
+            apply(op)
+
+            if isinstance(op, MeasureNode):
+                assert len(op.wires) == 1
+                # if a wire has already been updated we still
+                # want to map *from* the original wire
+                if name_and_wire in updated:
+                    new_wire = updated[name_and_wire] + 1
+                    wire_map[measured_wire] = new_wire
+                else:
+                    measured_wire = op.wires[0]
+                    new_wire = _find_new_wire(wires)
+                    wires += new_wire
+                    wire_map[measured_wire] = new_wire
+                updated[name_and_wire] = new_wire
```

Comment:
Up to you, but it wasn't clear to me why we needed the name in `name_and_wire` and if there could be any issues with converting `op.wires` to a string (e.g. if one of the wires is a function - admittedly this never happens).

I think we can have a similar functionality just by keeping a reverse dictionary:
```suggestion
    reverse_wire_map = {v: k for k, v in wire_map.items()}

    copy_ordered_ops = copy.deepcopy(ordered_ops)

    with QuantumTape() as tape:
        for _, op in copy_ordered_ops:
            new_wires = [wire_map[w] for w in op.wires]
            op._wires = Wires(new_wires)  # TODO: find a better way to update operation wires
            apply(op)

            if isinstance(op, MeasureNode):
                assert len(op.wires) == 1
                measured_wire = op.wires[0]

                new_wire = _find_new_wire(wires)
                wires += new_wire

                original_wire = reverse_wire_map[measured_wire]
                wire_map[original_wire] = new_wire
                reverse_wire_map[new_wire] = original_wire
```